nomenclature/processor/data_validator.py

```diff
@@ -276,6 +286,32 @@ def from_file(
 
         return cls(file=file, criteria_items=criteria_items, output_path=output_path)  # type: ignore
 
+    @classmethod
+    def from_definition(
+        cls, definition: DataStructureDefinition, output_path: Path | str | None = None
+    ) -> "DataValidator":
+        criteria_items = []
+
+        for name, variable in definition.variable.items():
+            if any([i in VALIDATION_ARGS for i in variable.extra_attributes]):
```

Comment:
Personal preference against since letter variable names:

```suggestion
            if any([attribute in VALIDATION_ARGS for attribute in variable.extra_attributes]):
```

Reply:
Done